[CI] GeoShapeWithDocValuesIT testMappingUpdate failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/qk5o6myo5rlsa/tests/:x-pack:plugin:spatial:internalClusterTest/org.elasticsearch.xpack.spatial.search.GeoShapeWithDocValuesIT/testMappingUpdate

**Reproduction line:**
`./gradlew ':x-pack:plugin:spatial:internalClusterTest' --tests "org.elasticsearch.xpack.spatial.search.GeoShapeWithDocValuesIT.testMappingUpdate" -Dtests.seed=644AA8F54C4A2DDC -Dtests.locale=ar-BH -Dtests.timezone=Asia/Yerevan -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.spatial.search.GeoShapeWithDocValuesIT&tests.test=testMappingUpdate

**Failure excerpt:**
```
junit.framework.AssertionFailedError: Unexpected exception type, expected IllegalArgumentException but got org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: using deprecated parameters [strategy] in mapper [shape] of type [geo_shape] is no longer allowed

  at __randomizedtesting.SeedInfo.seed([644AA8F54C4A2DDC:52041F053BA64CF4]:0)
  at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2746)
  at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2734)
  at org.elasticsearch.xpack.spatial.search.GeoShapeWithDocValuesIT.testMappingUpdate(GeoShapeWithDocValuesIT.java:67)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:834)

  Caused by: org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: using deprecated parameters [strategy] in mapper [shape] of type [geo_shape] is no longer allowed

    at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:303)
    at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.applyRequest(MetadataMappingService.java:110)
    at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.execute(MetadataMappingService.java:78)
    at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:745)
    at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:361)
    at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:224)
    at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:161)
    at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139)
    at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:678)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:259)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:222)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.lang.Thread.run(Thread.java:834)

    Caused by: java.lang.IllegalArgumentException: using deprecated parameters [strategy] in mapper [shape] of type [geo_shape] is no longer allowed

      at org.elasticsearch.xpack.spatial.index.mapper.GeoShapeWithDocValuesFieldMapper$TypeParser.parse(GeoShapeWithDocValuesFieldMapper.java:208)
      at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:204)
      at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:154)
      at org.elasticsearch.index.mapper.RootObjectMapper$TypeParser.parse(RootObjectMapper.java:149)
      at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:98)
      at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:92)
      at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:301)
      at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.applyRequest(MetadataMappingService.java:110)
      at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.execute(MetadataMappingService.java:78)
      at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:745)
      at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:361)
      at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:224)
      at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:161)
      at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:139)
      at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:177)
      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:678)
      at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:259)
      at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:222)
      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
      at java.lang.Thread.run(Thread.java:834)

```

--- a/x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/search/GeoShapeWithDocValuesIT.java
+++ b/x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/search/GeoShapeWithDocValuesIT.java
@@ -10,6 +10,7 @@ package org.elasticsearch.xpack.spatial.search;
 import org.elasticsearch.Version;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.mapper.MapperParsingException;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.search.geo.GeoShapeIntegTestCase;
 import org.elasticsearch.test.VersionUtils;
@@ -49,10 +50,10 @@ public class GeoShapeWithDocValuesIT extends GeoShapeIntegTestCase {
         return true;
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/77755")
     public void testMappingUpdate() {
         // create index
-        assertAcked(client().admin().indices().prepareCreate("test").setSettings(settings(randomSupportedVersion()).build())
+        Version version = randomSupportedVersion();
+        assertAcked(client().admin().indices().prepareCreate("test").setSettings(settings(version).build())
             .setMapping("shape", "type=geo_shape").get());
         ensureGreen();
 
@@ -65,9 +66,18 @@ public class GeoShapeWithDocValuesIT extends GeoShapeIntegTestCase {
             "  }\n" +
             "}";
 
-        IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> client().admin().indices()
-            .preparePutMapping("test")
-            .setSource(update, XContentType.JSON).get());
-        assertThat(e.getMessage(), containsString("mapper [shape] of type [geo_shape] cannot change strategy from [BKD] to [recursive]"));
+        if (version.before(Version.V_8_0_0)) {
+            IllegalArgumentException e = expectThrows(IllegalArgumentException.class, () -> client().admin().indices()
+                .preparePutMapping("test")
+                .setSource(update, XContentType.JSON).get());
+            assertThat(e.getMessage(),
+                containsString("mapper [shape] of type [geo_shape] cannot change strategy from [BKD] to [recursive]"));
+        } else {
+            MapperParsingException e = expectThrows(MapperParsingException.class, () -> client().admin().indices()
+                .preparePutMapping("test")
+                .setSource(update, XContentType.JSON).get());
+            assertThat(e.getMessage(),
+                containsString("using deprecated parameters [strategy] in mapper [shape] of type [geo_shape] is no longer allowed"));
+        }
     }
 }

